Code/GraphMol/ChemReactions/ReactionRunner.cpp

```diff
@@ -854,4 +882,192 @@ std::vector<MOL_SPTR_VECT> run_Reactants(const ChemicalReaction &rxn,
   return productMols;
 }  // end of ChemicalReaction::runReactants()
 
+// Generate a product based on a SINGLE reactant
```

Comment:
Oh god. The SMARTS spec actually allows you to specify multiple product templates? I just read up on the spec, apparently this is how they implement logical OR for multiple reactant templates

[CH3:7][CH3:7] >> [CH3:7][CH3:8]

What a mess. Is this how rdkit implements multiple reactant/product templates as well? eg something like:


[CH3:8][CH3:7] >> [CH3:7][CH3:7]